### ISSUE TITLE
CartoonGAN: Generative Adversarial Networks for Photo Cartoonization

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#9]
![image](https://user-images.githubusercontent.com/46308407/104114755-6bbaf800-534b-11eb-9c65-c3790ac31538.png)

# Key Idea

- Unpaired image-to-image translation using domain-specific GAN formulation.
- Explicitly train Discriminator to consider **edge** to achieve cartoon-like styles.
- Employ sparse l1 regularization for content loss to cope with huge pixel-wise distance due to style differences.
- Lightly pretrain the generator as *auto-encoder* as a means of initialization, prompting quicker and better convergence.

# Previous Approaches

### Image Stylization with NNs

- Neural style transfer
  - Transform a content image w.r.t. a single reference style image.
  - **Not effective** in cartoonization

- CNNMRF (CNN Markov Random Field)
  - Rely on local matching of CNN feature maps
  - Often results in semantically incorrect outputs (due to local matching)

### Image Synthesis with GANs

- Pix2Pix
  - Requires paired image samples
- CycleGAN
  - Training takes very long
  - Poor results on cartoon domain, due to specific characteristics (e.g. sharp edges, flat texture)

### Network Architecture

- Residual Blocks
  - Relieves gradient vanishing problem, enabling deeper models to be trained well
- Batch Normalization
  - Soothes problem of covariate shift and reduce oscillation in training. 
- LeakyReLU
  - Widely used activation on CNNs

# Formulation

![image](https://user-images.githubusercontent.com/46308407/104114764-77a6ba00-534b-11eb-86b3-2d3681f7f73e.png)

### Generator

- Auto-encoder-like architecture, with 8 residual blocks at the bottleneck.
- Final 7*7 convolution layer transforms feature maps back to RGB channels.

### Discriminator

- Deliberately use shallow network, since the task is less challenging compared to Generator.
- Takes in patches rather than full images, since cartoon-like styles are local quality.
- Employ LeakyReLU activations.

### Loss Function

- The core contribution of this paper.
- Consists of two parts, (1) adversarial loss and (2) content loss. (w=10 in experiment)

![image](https://user-images.githubusercontent.com/46308407/104114805-da985100-534b-11eb-89d0-acb021e71336.png)

#### (1) Adversarial Loss

![image](https://user-images.githubusercontent.com/46308407/104114858-69a56900-534c-11eb-8bc8-3ec2b7aa7ac5.png)

- Heuristically discovered clear edges are important features of cartoon-style paintings.
- For edge loss, create edge smoothed versions e_i from cartoon image c_i and encourage Discriminator to separate these edge-smoothed images from original images with sharp edges.
  - Use standard Canny edge detectors to find edge pixels
  - Dilate edge regions
  - Gaussian-smooth the regions

#### (2) Content Loss

![image](https://user-images.githubusercontent.com/46308407/104114864-745ffe00-534c-11eb-997a-d60145ad5d99.png)

- Content loss is measured by l1 distance between VGG feature maps.
- Even though certain feature maps capture the content part well, due to vast difference in styles, the l2 distance can still be large.
- Using sparse l1 distance helps remedy this problem.

### Initialization

- Pretrain the generator only with content loss for 10 epochs, which enables it to essentially **reconstruct** the input image.
- This initialization helps convergence.

# Experiments/Results

- Edge Generation

![image](https://user-images.githubusercontent.com/46308407/104114896-c43ec500-534c-11eb-8902-204561241860.png)

- Comparison with Neural-style-transfer and CycleGAN.

![image](https://user-images.githubusercontent.com/46308407/104114909-e59fb100-534c-11eb-80f4-ddb11ade1e97.png)

- Comparison with CNNMRF and Deep Analogy.

![image](https://user-images.githubusercontent.com/46308407/104114917-fa7c4480-534c-11eb-98d2-0c0f4d63972c.png)

- Ablation Study

![image](https://user-images.githubusercontent.com/46308407/104114920-0d8f1480-534d-11eb-9d7d-af482eab99ff.png)

# Comments

- Quantitative comparison이 전혀 없는 점이 조금 아쉽다. 생성 모델 전반의 문제이긴 하지만, 샘플 차원으로 제시하고 눈으로 확인하는 것은 cherry-picking issue가 클 것 같다는 느낌을 지우기 어렵다.
- Cartoonization이라는 구체적인 목표를 위해 Edge Generation을 별도로 구현한 점이 인상적. Im2Pencil도 그렇고, 이런 특수 분야의 stylization의 경우 Prior Domain 지식이 모델에 크게 반영되는 것 같다.
- Content Loss만으로 먼저 학습해서 잘 수렴할 수 있도록 한 점은 똑똑하다고 생각한다. 실제로 GAN은 학습하기 어려운 걸로 유명한데, 이런 initialization으로 학습 불안정성 문제를 (다른 task)에서도 경감할 수 있는지 궁금하다. 다른 논문들에서는 이렇게 하는 경우를 별로 못본 걸로 봐서는 WGAN, LSGAN, Spectral Normalization 등 좋은 기법들이 많이 등장해서 수렴이 꽤 잘 되기 때문일지, 아니면 이런 방식이 별다른 효과가 없기 때문일지 모르겠다.
- 2018년 논문이다보니 자연스럽게 Batch Normalization을 활용했는데, Adaptive Instance Normalization 논문을 보면 BN은 batch 내부의 통계량을 통일함으로서 style을 fuse, 해치는 경향이 있다고 한다. 여기에 instance normalization, 나아가 AdaIN을 적용하면 어떨지 궁금하다.
